core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -16,105 +16,94 @@
+  val remoteLogSegmentId = new RemoteLogSegmentId(idPartition, Uuid.randomUuid())
+  val rlsMetadata: RemoteLogSegmentMetadata = new RemoteLogSegmentMetadata(remoteLogSegmentId, baseOffset, lastOffset,
+    time.milliseconds(), brokerId, time.milliseconds(), segmentSize, Collections.singletonMap(0, 0L))
 
   @BeforeEach
   def setup(): Unit = {
-    val tempDir = Files.createTempDirectory("kafka-test-").toFile
-    tempDir.deleteOnExit()
-
-    offsetIndex = new OffsetIndex(new File(tempDir, "offset-index"), baseOffset, maxIndexSize = maxEntries * 8)
-    timeIndex = new TimeIndex(new File(tempDir, "time-index"), baseOffset = baseOffset, maxIndexSize = maxEntries * 12)
-    appendIndexEntries()
-
-    // Fetch indexes only once to build the cache, later it should be available in the cache
-    when(rsm.fetchIndex(any(classOf[RemoteLogSegmentMetadata]), ArgumentMatchers.eq(IndexType.OFFSET)))
-      .thenReturn(new FileInputStream(offsetIndex.file))
-
-    when(rsm.fetchIndex(any(classOf[RemoteLogSegmentMetadata]), ArgumentMatchers.eq(IndexType.TIMESTAMP)))
-      .thenReturn(new FileInputStream(timeIndex.file))
-
-    when(rsm.fetchIndex(any(classOf[RemoteLogSegmentMetadata]), ArgumentMatchers.eq(IndexType.TRANSACTION)))
-      .thenReturn(new FileInputStream(File.createTempFile("kafka-test-", ".txn", tempDir)))
-
-    cache = new RemoteIndexCache(remoteStorageManager = rsm, logDir = createLogDir)
-
-    val topicIdPartition = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0))
-    rlsMetadata = new RemoteLogSegmentMetadata(new RemoteLogSegmentId(topicIdPartition, Uuid.randomUuid()),
-      baseOffset, offsetIndex.lastOffset, -1L, 1, 1024, 1024,
-      Collections.singletonMap(0, 0L))
-  }
-
-  private def createLogDir = {
-    Files.createTempDirectory("kafka-").toString
-  }
-
-  private def appendIndexEntries(): Unit = {
-    val curTime = System.currentTimeMillis()
-    for (i <- 0 until offsetIndex.maxEntries) {
-      val offset = offsetIndex.baseOffset + i + 1
-      offsetIndex.append(offset, i)
-      timeIndex.maybeAppend(curTime + i, offset, skipFullCheck = true)
-    }
-
-    offsetIndex.flush()
-    timeIndex.flush()
+    Files.createDirectory(tpDir.toPath)
+    val txnIdxFile = new File(tpDir, "txn-index" + UnifiedLog.TxnIndexFileSuffix)
+    txnIdxFile.createNewFile()
+    when(rsm.fetchIndex(any(classOf[RemoteLogSegmentMetadata]), any(classOf[IndexType])))
+      .thenAnswer(ans => {
+        val metadata = ans.getArgument[RemoteLogSegmentMetadata](0)
+        val indexType = ans.getArgument[IndexType](1)
+        val maxEntries = (metadata.endOffset() - metadata.startOffset()).asInstanceOf[Int]
+        val offsetIdx = new OffsetIndex(new File(tpDir, String.valueOf(metadata.startOffset()) + UnifiedLog.IndexFileSuffix),
+          metadata.startOffset(), maxIndexSize = maxEntries * 8)
+        val timeIdx = new TimeIndex(new File(tpDir, String.valueOf(metadata.startOffset()) + UnifiedLog.TimeIndexFileSuffix),
```

Comment:
Why do we need to create these index files always even when the indextype is not meant for these?

Reply:
To append the index entries for offsetIdx and timeIdx. The entries are appended only once for each test.